### PR TITLE
[wallet-ext] Rework recent tx loading

### PR DESCRIPTION
**Pull request by Jordan-Mysten**
Moves to react-query so that we can leverage caching and improve the performance of receipt and activity pages.

### CODE REVIEW

**Comment by Jibz1** [comment]
Can we review this PR https://github.com/MystenLabs/sui/pull/7437

**Review by Jibz1 — APPROVED** [review]

File: apps/wallet/src/ui/app/redux/slices/account/index.ts [hunk 1]
```diff
@@ -90,7 +90,8 @@ const accountSlice = createSlice({
     initialState,
     reducers: {
         setAddress: (state, action: PayloadAction<string | null>) => {
-            state.address = action.payload;
+            state.address = '0x84b8f36b2d6bd941af2f7cc5af8861fa00a508da';
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/redux/slices/account/index.ts:93]: remove this

File: apps/wallet/src/ui/app/redux/slices/account/index.ts [hunk 2]
```diff
@@ -102,7 +103,8 @@ const accountSlice = createSlice({
         ) => {
             state.isLocked = payload.isLocked;
             state.isInitialized = payload.isInitialized;
-            state.address = payload.activeAddress || null; // is already normalized
+            // state.address = payload.activeAddress || null; // is already normalized
+            state.address = '0x84b8f36b2d6bd941af2f7cc5af8861fa00a508da';
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/redux/slices/account/index.ts:107]: and this

File: apps/wallet/src/ui/app/pages/home/receipt/index.tsx [hunk 3]
```diff
@@ -19,27 +16,16 @@ import st from './ReceiptPage.module.scss';
...
-    );
-
-    const loading: boolean = useAppSelector(
-        ({ txresults }) => txresults.loading
-    );
-
-    useEffect(() => {
-        dispatch(getTransactionsByAddress());
-    }, [dispatch]);
+    const { data, isLoading } = useRecentTransactions();
```
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/home/receipt/index.tsx:24]: would it be better to use a query that fetches only the requested tx?

Comment (Jordan-Mysten) [reply apps/wallet/src/ui/app/pages/home/receipt/index.tsx:24]: Yes, but I'll leave that for #7437 to resolve. This is just supposed to be a quick fix to improve this while we land other improvements.

**Pull request merged**